libsolidity/analysis/ReferencesResolver.cpp

```diff
@@ -296,11 +296,19 @@ void ReferencesResolver::endVisit(VariableDeclaration const& _variable)
 					{
 						typeLoc = DataLocation::Storage;
 						if (_variable.isLocalVariable())
-							m_errorReporter.warning(
-								_variable.location(),
-								"Variable is declared as a storage pointer. "
-								"Use an explicit \"storage\" keyword to silence this warning."
-							);
+						{
+							if (_variable.sourceUnit().annotation().experimentalFeatures.count(ExperimentalFeature::V050))
+								fatalTypeError(
```

Comment:
This should not be a fatal type error, because type checking can continue.

Reply:
Perhaps we should add a helper to `m_errorReporter` that turns warnings into errors if 0.5.0 is activated.

Reply:
Took fatal because the other case above does the same, though probably none of them must be fatal.

The helper I think wouldn't work given not all warnings are errors in 0.5.0.